aws-datastore/src/main/java/com/amplifyframework/datastore/AWSDataStorePlugin.java

```diff
@@ -86,33 +86,50 @@ private AWSDataStorePlugin(
+     * fully qualified class name of com.amplifyframework.datastore.generated.model.AmplifyModelProvider.
+     *
+     * This constructor will attempt to find that provide by means of reflection. If you have changed
+     * the path to your {@link ModelProvider} instance, and do not use the default path generated by
+     * the Amplify Code Generator, then this will be break. Likewise, if you need to provide a custom
+     * {@link ModelProvider}, this will not work. In both cases, you should prefer one of the other
+     * overloads such as {@link #AWSDataStorePlugin(ModelProvider)}.
+     *
+     * If remote synchronization is enabled, it will be performed via {@link Amplify#API}.
+     *
+     * @throws DataStoreException If it is not possible to access the code-generated model provider
+     */
+    @SuppressWarnings({"unused", "WeakerAccess"})
+    public AWSDataStorePlugin() throws DataStoreException {
+        this(ModelProviderLocator.locate(), Amplify.API);
+    }
+
+    /**
+     * Constructs an {@link AWSDataStorePlugin} which can warehouse the model types provided by
      * the supplied {@link ModelProvider}. If remote synchronization is enabled, it will be
      * performed through {@link Amplify#API}.
      * @param modelProvider Provider of models to be usable by plugin
-     * @return An {@link AWSDataStorePlugin} which warehouses the provided models
      */
-    @NonNull
-    @SuppressWarnings("WeakerAccess")
-    public static AWSDataStorePlugin forModels(@NonNull ModelProvider modelProvider) {
-        Objects.requireNonNull(modelProvider);
-        return create(modelProvider, Amplify.API);
+    @SuppressWarnings({"unused", "WeakerAccess"})
+    public AWSDataStorePlugin(@NonNull ModelProvider modelProvider) {
+        this(Objects.requireNonNull(modelProvider), Amplify.API);
     }
 
     /**
-     * Creates an {@link AWSDataStorePlugin} which can warehouse the model types provided by the
+     * Constructs an {@link AWSDataStorePlugin} which can warehouse the model types provided by the
      * supplied {@link ModelProvider}. If remote synchronization is enabled, it will be performed
      * through the provided {@link GraphQlBehavior}.
      * @param modelProvider Provides the set of models to be warehouse-able by this system
      * @param api Interface to a remote system where models will be synchronized
-     * @return An {@link AWSDataStorePlugin} which warehouses the provided model types
      */
-    @NonNull
-    public static AWSDataStorePlugin create(@NonNull ModelProvider modelProvider, @NonNull GraphQlBehavior api) {
-        Objects.requireNonNull(modelProvider);
-        Objects.requireNonNull(api);
-        ModelSchemaRegistry modelSchemaRegistry = ModelSchemaRegistry.instance();
-        return new AWSDataStorePlugin(modelSchemaRegistry, modelProvider, api);
+    @SuppressWarnings("WeakerAccess")
+    public AWSDataStorePlugin(@NonNull ModelProvider modelProvider, @NonNull GraphQlBehavior api) {
```

Comment:
on iOS we're treating this constructor as internal, only for dependency injection purposes. Customers cannot pass their own API implementation because of how strict our sync engine is when it comes to how API should behave.

Reply:
This is a good point. I'll look to remove the `public` access.